src/site/antora/modules/ROOT/pages/manual/webapp.adoc

```diff
@@ -15,311 +15,626 @@
-* IBM WebSphere 8.0 and higher
+include::partial$manual/dependencies-log4j-jakarta-web.adoc[]
 
-==== The Short Story
+If you are writing a Servlet 3.0 or later application, Apache Log4j Web will register a
+https://jakarta.ee/specifications/servlet/5.0/apidocs/jakarta/servlet/servletcontainerinitializer[`ServletContainerInitializer`]
+that takes care of configuring the Log4j lifecycle for you. Under the hood this will:
 
-Log4j 2 "just works" in Servlet 3.0 and newer web applications. It is capable of automatically starting when the application deploys and shutting down when the application undeploys. Thanks to the https://docs.oracle.com/javaee/6/api/javax/servlet/ServletContainerInitializer.html[ServletContainerInitializer] API added to Servlet 3.0, the relevant `Filter` and `ServletContextListener` classes can be registered dynamically on web application startup.
+* initialize Log4j Core with the correct configuration file,
+* register a `Log4jServletContextListener` to automatically shut down Log4j Core, when the application shuts down,
+* register a `Log4jServletFilter` to enable the xref:manual/lookups.adoc#WebLookup[web lookup].
 
-[warning]
-====
-Important Note!
-For performance reasons, containers often ignore certain JARs known not to contain TLDs or `ServletContainerInitializer`s and do not scan them for web-fragments and initializers. Importantly, Tomcat 7 <7.0.43 ignores all JAR files named log4j*.jar, which prevents this feature from working. This has been fixed in Tomcat 7.0.43, Tomcat 8, and later. In Tomcat 7 <7.0.43 you will need to change `catalina.properties` and remove "log4j*.jar" from the `jarsToSkip` property. You may need to do something similar on other containers if they skip scanning Log4j JAR files.
+See also <<web-application-specific>>.
+
+[WARNING]
 ====
+While the Servlet Specification allows web fragments to automatically add context listeners, it does not give any guarantees regarding the order in which those listeners are executed
+(cf. https://jakarta.ee/specifications/servlet/5.0/jakarta-servlet-spec-5.0#Assembling_the_descriptor[section 8.2.3]).
 
-==== The Long Story
+If other context listeners in your application use logging, you need to make sure that `Log4jServletContextListener` is the last listener to be executed at shutdown.
+In order to do it you must create a `web.xml` descriptor and add the `Log4jServletContextListener` explicitly as **first** context listener:
 
-The Log4j 2 Web JAR file is a web-fragment configured to order before any other web fragments in your application. It contains a `ServletContainerInitializer` (`Log4jServletContainerInitializer`) that the container automatically discovers and initializes. This adds the `Log4jServletContextListener` and `Log4jServletFilter` to the `ServletContext`. These classes properly initialize and deinitialize the Log4j configuration.
+[source,xml,indent=0]
+----
+include::example$manual/webapp/web.xml[tag=context-listener]
```

Comment:
Mind providing a link to the actual source in the caption of the source block, please?

Reply:
Fixed in 2809eaae3db7089f9c56b9b1e5a218e0606a4f38.